pkg/daemon/ceph/client/pool.go

```diff
@@ -326,18 +336,112 @@ func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
-	failureDomain := pool.FailureDomain
-	if failureDomain == "" {
-		failureDomain = cephv1.DefaultFailureDomain
+func createStretchedReplicationCrushRule(context *clusterd.Context, clusterInfo *ClusterInfo, ruleName string, pool cephv1.PoolSpec) error {
+	// Get the current CRUSH map
+	var crushMap CrushMap
+	crushMap, err := GetCrushMap(context, clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get crush map")
+	}
+
+	// Fetch the compiled crush map
+	compiledCRUSHMapFilePath, err := GetCompiledCrushMap(context, clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get crush map")
+	}
+
+	// Decompile the plain text to CRUSH binary format
+	err = decompileCRUSHMap(context, compiledCRUSHMapFilePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to compile crush map")
+	}
+
+	// Build plain text rule
+	plainRule := buildStretchClusterPlainCrushRule(crushMap, ruleName, pool)
+
+	// Append plain rule to the decompiled crush map
+	decompiledCRUSHMapFilePath := buildDecompileCRUSHFileName(compiledCRUSHMapFilePath)
+	f, err := os.OpenFile(filepath.Clean(decompiledCRUSHMapFilePath), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0400)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open decompiled crush map %q", decompiledCRUSHMapFilePath)
+	}
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			logger.Errorf("failed to close file %q. %v", f.Name(), err)
+		}
+		for _, file := range []string{compiledCRUSHMapFilePath, decompiledCRUSHMapFilePath, buildCompileCRUSHFileName(decompiledCRUSHMapFilePath)} {
+			err := os.Remove(file)
+			if err != nil {
+				logger.Errorf("failed to remove file %q. %v", file, err)
+			}
+		}
+	}()
+	if _, err := f.WriteString(plainRule); err != nil {
+		return errors.Wrapf(err, "failed to append replicated plain crush rule to decompiled crush map %q", decompiledCRUSHMapFilePath)
 	}
 
-	// set the crush root to the default if not already specified
-	crushRoot := "default"
-	if pool.CrushRoot != "" {
-		crushRoot = pool.CrushRoot
+	// Compile the plain text to CRUSH binary format
+	err = compileCRUSHMap(context, decompiledCRUSHMapFilePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to compile crush map")
 	}
-	args := []string{"osd", "crush", "rule", "create-replicated", ruleName, crushRoot, failureDomain}
+
+	// Inject the new CRUSH Map
+	err = injectCRUSHMap(context, clusterInfo, buildCompileCRUSHFileName(decompiledCRUSHMapFilePath))
+	if err != nil {
+		return errors.Wrap(err, "failed to inject crush map")
+	}
+
+	return nil
+}
+
+// TODO: use me once https://tracker.ceph.com/issues/7770 is implemented
+// And remove the other version of createStretchedReplicationCrushRule()
+// Due to timing issue, we needed to implement a quick solution to get multisite in time for 1.5
+// So we came up with the "append in file" solution obviously not so elegant
```

Comment:
It could be a while before we get the json implementation. How about if we keep a simpler comment in the other function instead of keeping this entire function commented out? Commented code always goes stale very quickly and it won't be complex to re-implement. Also, it will also create a dependency on a version of ceph, so we would need the plain text crush compilation even longer for older versions of ceph.